game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculator.java

```diff
@@ -144,6 +141,15 @@ public AggregateResults calculate(
     }
   }
 
+  private Collection<Unit> mergeUnitCollections(Collection<Unit> c1, Collection<Unit> c2) {
+    var combined = new HashSet<>(c1);
+    combined.addAll(c2);
+    // Check that the two collections were distinct and didn't have duplicates. This helps catch
+    // logic errors in AI code that would otherwise be hard to debug.
+    Preconditions.checkState(combined.size() == c1.size() + c2.size());
```

Comment:
:+1:  This is an excellent use of a 'state' check

Suggestion: Consider moving the comment on L147 & L148 into the `checkState` method. IIRC there is an API overload on `checkState` that allows for an error message. Further, the error message could be enhanced by printing out the contents of the two sets for maximum debug-ability if we were ever to see this problem.

Reply:
Done.